src/main/java/org/elasticsearch/plugins/PluginManager.java

```diff
@@ -19,674 +19,557 @@
+                    case "-timeout":
+                        replacement = "--timeout";
+                        break;
+                    case "-i":
+                    case "--install":
+                    case "-install":
+                        replacement = "install";
+                        break;
+                    case "-r":
+                    case "--remove":
+                    case "-remove":
+                    case "remove":
+                        replacement = "uninstall";
+                        break;
+                    case "-l":
+                    case "--list":
+                        replacement = "list";
+                        break;
+                }
+
+                if (replacement != null) {
+                    terminal.printError("WARN: option [%s] has been deprecated by [%s]. Please update your scripts.", current, replacement);
+                    args[i] = replacement;
                 }
-        };
-
-        // Install the all-trusting trust manager
-        try {
-            SSLContext sc = SSLContext.getInstance("SSL");
-            sc.init(null, trustAllCerts, new java.security.SecureRandom());
-            HttpsURLConnection.setDefaultSSLSocketFactory(sc.getSocketFactory());
-        } catch (Exception e) {
-            throw new ElasticsearchException("Failed to install all-trusting trust manager", e);
+            }
         }
     }
 
-    public void downloadAndExtract(String name) throws IOException {
-        if (name == null) {
-            throw new ElasticsearchIllegalArgumentException("plugin name must be supplied with --install [name].");
-        }
-        HttpDownloadHelper downloadHelper = new HttpDownloadHelper();
-        boolean downloaded = false;
-        HttpDownloadHelper.DownloadProgress progress;
-        if (outputMode == OutputMode.SILENT) {
-            progress = new HttpDownloadHelper.NullProgress();
-        } else {
-            progress = new HttpDownloadHelper.VerboseProgress(SysOut.getOut());
+    /**
+     * List all installed plugins
+     */
+    static class List extends CliTool.Command {
+
+        private static final String NAME = "list";
+
+        private static final CliToolConfig.Cmd CMD = cmd(NAME, List.class).build();
+
+        public static Command parse(Terminal terminal, CommandLine cli) {
+            return new List(terminal);
         }
 
-        if (!Files.isWritable(environment.pluginsFile())) {
-            throw new IOException("plugin directory " + environment.pluginsFile() + " is read only");
+        List(Terminal terminal) {
+            super(terminal);
         }
 
-        PluginHandle pluginHandle = PluginHandle.parse(name);
-        checkForForbiddenName(pluginHandle.name);
+        @Override
+        public ExitStatus execute(Settings settings, Environment env) throws Exception {
+            Collection<PluginDescriptor> plugins = installedPluginsRepository(env, settings).list();
+            if (plugins != null) {
```

Comment:
Looking at the code, `plugins` should never be `null`. Not sure we need to test it.